lib/srv/sess.go

```diff
@@ -1160,6 +1160,18 @@ func (s *session) sessionRecordingMode() string {
 	return sessionRecMode
 }
 
+// roleSessionRecordingMode returns the session recording mode defined for SSH
+// sessions.
+func (s *session) roleSessionRecordingMode() constants.SessionRecordingMode {
+	// When recording mode is sync, any failure should abort the session to
+	// honor the definition.
+	if services.IsRecordSync(s.scx.SessionRecordingConfig.GetMode()) {
+		return constants.SessionRecordingModeStrict
+	}
+
+	return s.scx.Identity.AccessChecker.SessionRecordingMode(constants.SessionRecordingServiceSSH)
```

Comment:
is `s.scx.SessionRecordingConfig.GetMode())` coming from static config? So it has a higher priority than the role?

Reply:
It is a cluster config.

> So it has a higher priority than the role?

Yes, we enforce strict mode when the cluster configuration is set to any sync recording (`node-sync` or `proxy-sync`). Overwriting the role value.

Reply:
That might not be the case when we [add the `off` option](https://github.com/gravitational/teleport/pull/40170), but we can discuss it there.